Docs: update the description of `ingress_application_gateway` of `azurerm_kubernetes_cluster` (#22875)
hashicorp/terraform-provider-azurerm: website/docs/r/kubernetes_cluster.html.markdown

* `ingress_application_gateway` - (Optional) A `ingress_application_gateway` block as defined below.

> -> **Note:** As the Application Gateway is deployed inside a Virtual Network, so the users (and Service Principals) that operate Application Gateways also have at least `Microsoft.Network/virtualNetworks/subnets/join/action` permission on the Virtual Network or Subnet. For more details, please visit [Virtual Network Permission](https://learn.microsoft.com/en-us/azure/application-gateway/configuration-infrastructure#virtual-network-permission).

Reviewer: ```suggestion
-> **Note:** Since the Application Gateway is deployed inside a Virtual Network, users (and Service Principals) that are operating the Application Gateway must have the `Microsoft.Network/virtualNetworks/subnets/join/action` permission on the Virtual Network or Subnet. For more details, please visit [Virtual Network Permission](https://learn.microsoft.com/en-us/azure/application-gateway/configuration-infrastructure#virtual-network-permission).
```

Author: Updated
